types/transaction.go

```diff
@@ -83,7 +83,16 @@ func (t *Transaction) ComputeHash(blockNumber uint64) *Transaction {
 
 func (t *Transaction) Copy() *Transaction {
 	tt := new(Transaction)
-	*tt = *t
+	tt.Nonce = t.Nonce
```

Comment:
Is there any difference with `*tt = *t` ?

Reply:
I've changed this because of a warning regarding copying this way the `size` field (`size atomic.Pointer[uint64]`)
```go
*tt = *t
```

This is the warning being reported: 
> assignment copies lock value to *tt: github.com/0xPolygon/polygon-edge/types.Transaction contains sync/atomic.Pointer[uint64] contains sync/atomic.noCopy